server/routes/utils/authentication.js

```diff
@@ -0,0 +1,26 @@
+const Utility = require(`${serverRoot}/lib/Utility`);
+
+const authentication = (req, res, next) => {
+  try {
+    let authCheck = false;
+    if (typeof req.session !== 'undefined') {
+      if (typeof req.session.data.signin_info !== 'undefined') {
+        if (req.session.data.signin_info.signed_in === 1) {
+          authCheck = true;
+        }
+      }
+    }
+    if (!authCheck) {
+      throw new Error('User not signed in');
```

Comment:
Why throw an Error just to catch it in the same location?

Reply:
Can `req.session.data.signin_info.signed_in` throw an exception if its undefined and this handes this and the authcheck failing?

Reply:
The error is thrown purely for logging purposes. With regards to your other question, I think it could be simplified, how are you thinking it should look?

Reply:
If you just did `    Utility.logException(new Error('User not signed in'));` then we wouldn't need to make the compiler rewind the stack.
I was trying to understand if there are any of cases where that catch block could be hit.

Reply:
We still want the user to be redirected if they are not signed in. Which is why we throw, so then in the catch it can redirect. However, we probably don't want to output an error, because it is not really an error and we'll pollute the logs. I also realised, if there is an actual error, that means something is wrong with the app and we should display an error page. I can change it to this if you agree:
const Utility = require(`${serverRoot}/lib/Utility`);

const authentication = (req, res, next) => {
  try {
    let authCheck = false;
    if (typeof req.session !== 'undefined') {
      if (typeof req.session.data.signin_info !== 'undefined') {
        if (req.session.data.signin_info.signed_in === 1) {
          authCheck = true;
        }
      }
    }
    if (!authCheck) {
      Utility.logInfo(err);
      return res.redirect(302, redirectUrl());
    } else {
      next();
    }
  } catch (err) {
    Utility.logException(err);
    next(err);
  }
};
const redirectUrl = () => {
    const port = req.socket.localPort && req.socket.localPort !== '80' ? ':' + req.socket.localPort : '';
    const returnUrl = `${req.protocol}://${req.hostname}${port}${req.originalUrl}`;
    return ${process.env.ACCOUNT_URL}/signin?return_to=${returnUrl}
}
module.exports = authentication;